docker/app/src/Controller/giftcard_controller.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace App\Controller;
+
+use Google\Cloud\Firestore\FirestoreClient;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+class giftcard_controller extends AbstractController
+{
+    /** 
+     *@Route("/giftcard/create", methods={"POST"})
+     */
+    public function giftcreate(Request $request): JsonResponse
+    {
+        $data = $request->toArray();
+        $projectId = 'matejgrubisicgiftcard';
+
+        $randomid = uniqid();
+
+        $type = "DIGITAL";
+        $time = getdate();
+        $amount = random_int(10,100);
+        $valid = true;
+        $sales = 'asjdirjgeije4394839fdf';
+
+        $firestore = new FirestoreClient([
+            'projectId' => $projectId,
+        ]);
+        
+        $document = $firestore->collection($_ENV['COLLECTION'])->newDocument();
+        $document->create($data);
+
+        
+
+        return new JsonResponse(
+            $document->snapshot()->data()
+        );
+        #works 
+    }
+
+    /** 
+     *@Route("/giftcard/search/{id}", methods={"GET"})
+     */
+    public function giftget($id): JsonResponse
+    {
+        $projectId = 'matejgrubisicgiftcard';
```

Comment:
I can see that projectId is repeating, you can add it as a Constant for this class, or for whole service